### PR TITLE
Added runners for enabling concurrent segment search via settings

### PR DESCRIPTION
**Pull request by martin-gaievski**
### Description
Added EnableConcurrentSegmentSearch runner, this will be used in semantic search workloads like https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/noaa_semantic_search or one that is in PR https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/342 

### Issues Resolved
building block for https://github.com/opensearch-project/opensearch-benchmark-workloads/issues/341

### Testing
- [x] New functionality includes testing

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: osbenchmark/worker_coordinator/runner.py [hunk 1]
```diff
@@ -2696,3 +2697,17 @@ async def __call__(self, opensearch, params):
 
     def __repr__(self, *args, **kwargs):
         return "deploy-ml-model"
+
+class EnableConcurrentSegmentSearch(Runner):
```
Comment (VijayanB) [inline osbenchmark/worker_coordinator/runner.py:2701]: nit: What do you think about naming as UpdateConcurrentSegmentSearchClusterSettings?

**Comment by VijayanB** [comment]
@martin-gaievski Do you have plan to add slice settings as well?

**Comment by martin-gaievski** [comment]
> @martin-gaievski Do you have plan to add slice settings as well?

do you mean setting `search.concurrent.max_slice_count`? I can add it to, it can be operation UpdateConcurrentSegmentSearchClusterSettings and two params: `enable` and `max_slice_count`.

File: osbenchmark/worker_coordinator/runner.py (hunk 1 above)
Comment (martin-gaievski) [reply osbenchmark/worker_coordinator/runner.py:2701]: ack, makes sense if we're adding the `slice count` param

**Review by VijayanB — APPROVED** [review]
LGTM

**Comment by VijayanB** [comment]
@IanHoang @gkamat Can you take a look at this PR? Thanks.

**Pull request merged**